toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -1197,7 +1268,8 @@ func (g *PkgGraph) DeepCopy() (deepCopy *PkgGraph, err error) {
 
 // MakeDAG ensures the graph is a directed acyclic graph (DAG).
 // If the graph is not a DAG, this routine will attempt to resolve any cycles to make the graph a DAG.
-func (g *PkgGraph) MakeDAG() (err error) {
+
+func (g *PkgGraph) MakeDAG(resolveCylesFromUpstream bool, outDir, tmpDir, workerTar, existingRpmsDir, existingToolchainRpmsDir string, usePreviewRepo, ignoreVersionToResolveSelfDep bool, repoFiles []string) (err error) {
 	timestamp.StartEvent("convert to DAG", nil)
```

Comment:
I was thinking for a while about how to avoid having to push this long list of arguments down the stack and prevent having to populate these new arguments with default values for when we don't want to use the upstream repo. Here's what I think we could do to remedy this issue:

1. Move these arguments to members of the `PkgGraph` struct. This way they are we don't have to include them in every function call in the stack.
2. Either:
    1.  add a new `EnableUpstreamCyclesResolution` method, which will accept the new arguments and set the internal members of the `PkgGraph` struct or
    2. update `NewPkgGraph` to accept the new arguments and set them as part of initialization.

Personally I'm a bigger fan of the new method, because that way every place in the code, which doesn't want to use the upstream remains unchanged and doesn't have to pass a bunch of empty strings and other unused arguments whenever they want to resolve cycles.

Alternatively to `EnableCyclesResolutionFromUpstream` we could also introduce a new "constructor" like `NewPkgGraphWithUpstreamCyclesResolution`.

Reply:
And to be completely fair, it would probably be even better to extract the responsibility of `PkgGraph` to resolve cycles into a separate class, which then operates on the graph, but that's a larger refactoring task for a completely different time.

Reply:
initializing the cloner early on inside the grapher instead of sending all the args required to create cloner. Making use of the new ConstructClonerWithNetwork